thrift-parser.js

```diff
@@ -366,6 +410,34 @@ module.exports = (buffer, offset = 0) => {
     return result;
   };
 
+  const readUnion = () => {
+    let subject = readKeyword('union');
+    let name = readName();
+    let items = readUnionBlock();
```

Comment:
Typically this results in a warning being emitted from the `thrift` binary because it is invalid syntax.  I have some other work that logs a debug warning if the thrift file has the "required" keyword, but it still deletes the property since it is invalid syntax (whereas "optional" is not invalid).  I'll get some more feedback on my end.